src/generate_module.ts

```diff
@@ -0,0 +1,81 @@
+/// <reference path="../typings/node/node.d.ts"/>
+/// <reference path="../typings/xml2js/xml2js.d.ts"/>
+/// <reference path="../typings/es6-promise/es6-promise.d.ts"/>
+/// <reference path="./parsers.ts"/>
+/// <reference path="./generators.ts"/>
+import fs = require("fs");
+import xml2js = require("xml2js");
+import promise = require("es6-promise");
+import util = require("util");
+
+import parsers = require("parsers");
+import generators = require("generators");
+
+var Promise = promise.Promise;
+
+function readFile(fileName: string) {
+    return new Promise((resolve, reject) => {
+        fs.readFile(fileName, function (err, data) {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(data.toString("utf8"));
+            }
+        });
+    });
+}
+
+function parseXml(xmlData: string) {
+    return new Promise((resolve, reject) => {
+        var parser = new xml2js.Parser();
+        parser.parseString(xmlData, function (err, result) {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(result);
+            }
+        });
+    });
+}
+
+function getDefinition(moduleName: string) {
+    var fileName = "./" + moduleName + ".xml";
+
+    return readFile(fileName).then(parseXml, console.log)
```

Comment:
Remember, `console.log` won't work in such way in browser. It should be bound to the `console` object first. As an explanation - compare outputs of `[1, 2, 3].map(console.log)` and `[1, 2, 3].map(function (x) { console.log(x); })` in browser console.

_(But it works in nodejs so we shouldn't care for now.)_